### PR TITLE
enh(api): implements infra and application layer for find CG use case

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to implements infrastructure and application layer for find contact groups use case

**Fixes** MON-12599

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/php/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroupsTest.php [hunk 1]
```diff
@@ -122,3 +127,29 @@
...
+        ->willReturn([$contactGroup]);
+
+    $this->user
+        ->expects($this->once())
+        ->method('isAdmin')
+        ->willReturn(true);
+
+    $presenter = new FindContactGroupsPresenterStub($this->presenterFormatter);
+    $useCase($presenter);
+    dump($presenter);
```
Comment (kduret) [inline tests/php/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroupsTest.php:147]: remove debug

File: src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+        $contactGroups = [];
+        while ($statement !== false && is_array($result = $statement->fetch(\PDO::FETCH_ASSOC))) {
+            $contactGroups[] = DbContactGroupFactory::createFromRecord($result);
+        }
+
+        return $contactGroups;
+    }
+
+    public function findAllByUserId(int $userId): array
```
Comment (kduret) [inline src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php:103]: missing phpdoc

File: tests/php/Core/Contact/Infrastructure/Api/FindContactGroups/FindContactGroupsControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+    namespace Tests\Core\Contact\Infrastructure\Api\FindContactGroups;
```
Comment (kduret) [inline tests/php/Core/Contact/Infrastructure/Api/FindContactGroups/FindContactGroupsControllerTest.php:24]:
```suggestion
namespace Tests\Core\Contact\Infrastructure\Api\FindContactGroups;
```

File: src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php [hunk 4]
```diff
@@ -65,7 +68,7 @@ public function __invoke(FindContactGroupsPresenterInterface $presenter): void
                 }
                 $contactGroups = $this->repository->findAllByUserId($this->user->getId());
             }
-        } catch (\Throwable $ex) {
+        } catch (\Throwable) {
```
Comment (jeremyjaouen) [inline src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php:71]: restore getTraceAsString in our logs

File: tests/php/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroupsTest.php [hunk 5]
```diff
@@ -122,3 +127,28 @@
     $presenter = new FindContactGroupsPresenterStub($this->presenterFormatter);
     $useCase($presenter);
 });
+
```
Comment (callapa) [inline tests/php/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroupsTest.php:130]: There is no test to check the error message

**Review by callapa — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**